### PR TITLE
[Feature](map-type) Support stream load and fix some bugs for map type

### PR DESCRIPTION
**Pull request by amorynan**
# Proposed changes

1. support stream load with json, csv format for map
2. fix olap convertor when compaction action in map column which has null 
3. support select outToFile for map
4. add some regression-test

#16547

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/data_types/data_type_factory.cpp [hunk 1]
```diff
@@ -170,8 +170,8 @@ DataTypePtr DataTypeFactory::create_data_type(const TypeDescriptor& col_desc, bo
     case TYPE_MAP:
         DCHECK(col_desc.children.size() == 2);
         nested = std::make_shared<vectorized::DataTypeMap>(
-                create_data_type(col_desc.children[0], col_desc.contains_nulls[0]),
-                create_data_type(col_desc.children[1], col_desc.contains_nulls[1]));
+                create_data_type(col_desc.children[0], false),
```
Comment (xy720) [inline be/src/vec/data_types/data_type_factory.cpp:173]: Why false? It seems should be true

**Review by xy720 — COMMENTED** [review]
Please add some regression test

File: be/src/vec/data_types/data_type_map.cpp [hunk 2]
```diff
@@ -91,57 +162,51 @@ Status DataTypeMap::from_string(ReadBuffer& rb, IColumn* column) const {
...
+                return Status::InvalidArgument("Cannot read map value from text '{}'",
+                                               value_element.to_string());
             }
-
-            size_t k_len = 0;
-            auto k_rb = rb.position();
-            while (kv_len > 0 && *k_rb != ':') {
-                k_len++;
-                k_rb++;
+            ReadBuffer krb(const_cast<char*>(key_element.data), key_element.size);
```
Comment (xy720) [inline be/src/vec/data_types/data_type_map.cpp:194]: Add support for null value.

File: regression-test/suites/map_p0/test_map_load_and_function.groovy [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        path.delete();
+    }
+    if (!path.exists()) {
+        assert path.mkdirs()
+    }
+    sql """
+                SELECT * FROM ${testTable} INTO OUTFILE "file://${outFilePath}/";
+    """
+    File[] files = path.listFiles()
+    assert files.length == 1
```
Comment (xiaokang) [inline regression-test/suites/map_p0/test_map_load_and_function.groovy:89]: should check file content

File: be/src/vec/olap/olap_data_convertor.cpp [hunk 4]
```diff
@@ -786,6 +786,12 @@ Status OlapBlockDataConvertor::OlapColumnDataConvertorArray::convert_to_olap(
     return Status::OK();
 }
 
+void OlapBlockDataConvertor::OlapColumnDataConvertorMap::set_source_column(
```
Comment (xiaokang) [inline be/src/vec/olap/olap_data_convertor.cpp:789]: Is necessary to override set_source_column and just call parent's set_source_column as impl?

File: be/src/vec/exprs/vexpr.cpp [hunk 5]
```diff
@@ -371,6 +372,15 @@ FunctionContext::TypeDesc VExpr::column_type_to_type_desc(const TypeDescriptor&
...
         break;
+    case TYPE_MAP:
+        CHECK(type.children.size() == 2);
+        // only support map key is scalar
+        CHECK(!type.children[0].is_complex_type());
+        out.type = FunctionContext::TYPE_MAP;
+        for (const auto& t : type.children) {
+            out.children.push_back(VExpr::column_type_to_type_desc(t));
+        }
+        break;
```
Comment (xiaokang) [inline be/src/vec/exprs/vexpr.cpp:383]: is TYPE_STRUCT also need to be handled? @xy720

**Review by xy720 — DISMISSED** [review]
LGTM

File: be/src/vec/exprs/vexpr.cpp (hunk 5 above)
Comment (xy720) [reply be/src/vec/exprs/vexpr.cpp:383]: Yes. It should be also support for struct.
It is used for function context.

**Review by xy720 — COMMENTED** [review]
The regression test need to be changed.

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**